### PR TITLE
improve `$header` modifier description

### PR DESCRIPTION
**Pull request by stanislav-atr**

### CODE REVIEW

File: docs/general/ad-filtering/create-own-filters.md [hunk 1]
```diff
@@ -445,9 +445,15 @@ The modifier `":" h_value` part may be omitted. In that case the modifier matche
 * `@@||example.com^$header=set-cookie:/foo\, bar\$/` unblocks requests whose responses have the `Set-Cookie` header with value matching the `foo, bar$` regular expression.
 * `@@||example.com^$header=set-cookie` unblocks requests whose responses have a `Set-Cookie` header with any value.
 
+:::caution Limitations
+
+In Adguard Browser Extension, [`$header`](#header-modifier) modifier is only compatible with [`$csp`](#csp-modifier) and [`$removeheader`](#removeheader-modifier) advanced modifiers. Rules that use [`$header`](#header-modifier) modifier with one of these, are subject to allowlisting logic of advanced modifiers.
```
Comment (ameshkov) [inline docs/general/ad-filtering/create-own-filters.md:450]: We should explain the difference in matching logic of this modifier. It is matched only when headers are received so if the request was blocked or redirected on an earlier stage nothing will be applied. This logic is similar for both the browser extension and CoreLibs.

File: docs/general/ad-filtering/create-own-filters.md [hunk 2]
```diff
@@ -445,9 +445,15 @@ The modifier `":" h_value` part may be omitted. In that case the modifier matche
...
+
+:::
+
 :::info Compatibility
 
-Rules with the `$header` modifier are supported by AdGuard for Windows, Mac, and Android, **running CoreLibs version 1.11 or later**.
+Rules with the `$header` modifier are supported by AdGuard for Windows, Mac, and Android, **running CoreLibs version 1.11 or later**, and AdGuard Browser Extension, **running TSUrlFilter vX.X.X or later**.
```
Comment (ameshkov) [inline docs/general/ad-filtering/create-own-filters.md:456]: So?

**Review by ameshkov — CHANGES_REQUESTED** [review]

File: docs/general/ad-filtering/create-own-filters.md (hunk 2 above)
Comment (stanislav-atr) [reply docs/general/ad-filtering/create-own-filters.md:456]: This is waiting for an actual version num on tsurlfilter 2.2 release

File: docs/general/ad-filtering/create-own-filters.md [hunk 3]
```diff
@@ -491,9 +490,22 @@ The modifier part, `":" h_value`, may be omitted. In that case, the modifier mat
 - `@@||example.com^$header=set-cookie:/foo\, bar\$/` unblocks requests whose responses have the `Set-Cookie` header with value matching the `foo, bar$` regular expression.
 - `@@||example.com^$header=set-cookie` unblocks requests whose responses have a `Set-Cookie` header with any value.
 
+##### `$header` modifier limitations {#header-modifier-limitations}
+
+:::caution Restrictions
+
+1. The `$header` modifier can be matched only when headers are received.
+  So if the request is blocked or redirected on an earlier stage, the modifier cannot be applied.
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:498]:
```suggestion
  So if the request is blocked or redirected at an earlier stage, the modifier cannot be applied.
```

File: docs/general/ad-filtering/create-own-filters.md [hunk 4]
```diff
@@ -565,16 +577,24 @@ AdGuard will try to close the browser tab with any address that matches a blocki
...
 
-:::info Compatibility
+##### `$popup` modifier limitations {#popup-modifier-limitations}
+
+:::caution Limitations
 
-- `$popup` modifier works best in AdGuard Browser Extension.
-- In AdGuard for Safari and iOS, `$popup` rules simply block the page right away.
-- In AdGuard for Windows, Mac, and Android, `$popup` modifier may not detect a popup in some cases and it will not be blocked.
+1. `$popup` modifier works best in AdGuard Browser Extension.
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:584]:
```suggestion
1. The `$popup` modifier works best in AdGuard Browser Extension.
```

File: docs/general/ad-filtering/create-own-filters.md [hunk 5]
```diff
@@ -565,16 +577,24 @@ AdGuard will try to close the browser tab with any address that matches a blocki
...
+1. In AdGuard for iOS and AdGuard for Safari, `$popup` rules simply block the page right away.
+1. In AdGuard for Windows, AdGuard for Mac, and AdGuard for Android, `$popup` modifier may not detect a popup
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:586]:
```suggestion
1. In AdGuard for Windows, AdGuard for Mac, and AdGuard for Android, the `$popup` modifier may not detect a popup
```

File: docs/general/ad-filtering/create-own-filters.md [hunk 6]
```diff
@@ -565,16 +577,24 @@ AdGuard will try to close the browser tab with any address that matches a blocki
...
+  in some cases and it will not be blocked.
   `$popup` modifier applies the `document` content type with a special flag which is passed to a blocking page.
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:588]:
```suggestion
  Thу `$popup` modifier applies the `document` content type with a special flag that is passed to a blocking page.
```

File: docs/general/ad-filtering/create-own-filters.md [hunk 7]
```diff
@@ -565,16 +577,24 @@ AdGuard will try to close the browser tab with any address that matches a blocki
...
   Blocking page itself can do some checks and close the window if it is really a popup.
   Otherwise, page should be loaded.
   It can be combined with other request type modifiers, such as `$third-party` and `$important`.
-- Rules with `$popup` modifier are not supported by AdGuard Content Blocker.
+
+:::
+
+:::info Compatibility
+
+Rules with `$popup` modifier are not supported by AdGuard Content Blocker.
```
Comment (el-termikael) [inline docs/general/ad-filtering/create-own-filters.md:597]:
```suggestion
Rules with the `$popup` modifier are not supported by AdGuard Content Blocker.
```

**Review by el-termikael — CHANGES_REQUESTED** [review]

**Review by el-termikael — APPROVED** [review]

**Pull request merged**